ui/src/notebooks/components/Pipe.tsx

```diff
@@ -0,0 +1,17 @@
+import {FC, createElement, useContext} from 'react'
+import {NotebookContext} from 'src/notebooks/context/notebook'
+
+import {PIPE_DEFINITIONS, PipeProp} from 'src/notebooks'
+
+const Pipe: FC<PipeProp> = ({idx}) => {
+  const {pipes} = useContext(NotebookContext)
+
+  if (!PIPE_DEFINITIONS.hasOwnProperty(pipes[idx].type)) {
+    throw new Error(`Pipe type [${pipes[idx].type}] not registered`)
+    return null
```

Comment:
Looks like this return is not going to be reached.

Reply:
it will if the data schema provides an invalid type, which shows up as a hydration problem. currently you are correct as we are only adding things that are already registered, but i'm expecting that to change as we migrate to new types and storing this on the server, and tracking this error is how we'll signal that we need to have a more graceful migration strategy that doesn't just leave types on the floor.